src/JBDelegateMetadataLib.sol

```diff
@@ -74,4 +74,91 @@ library JBDelegateMetadataLib {
         }
     }
 
+
+
+    /**
+     * @notice Add a delegate to an existing metadata
+     *
+     * @param _idToAdd         The id of the delegate to add
+     * @param _dataToAdd       The metadata of the delegate to add
+     * @param _originalMetadata The original metadata
+     *
+     * @return _newMetadata    The new metadata with the delegate added
+     */
+    function addToMetadata(bytes4 _idToAdd, bytes calldata _dataToAdd, bytes calldata _originalMetadata) public pure returns (bytes memory _newMetadata) {
+        // Get the first data offset - upcast to avoid overflow (same for other offset)...
+        uint256 _firstOffset = uint8(_originalMetadata[RESERVED_SIZE + ID_SIZE]);
+
+        // ...go back to the beginning of the previous word (ie the last word of the table, as it can be padded)
+        uint256 _lastWordOfTable = _firstOffset - 1;
+
+        // The last offset stored in the table and its index
+        uint256 _lastOffset;
+
+        uint256 _lastOffsetIndex;
+
+        // The number of words taken by the last data stored
+        uint256 _numberOfWordslastData;
+
+        // Iterate to find the last entry of the table, _lastOffset - we start from the end as the first value encountered
+        // will be the last offset
+        for(uint256 _i = _firstOffset * WORD_SIZE - 1; _i > _lastWordOfTable * WORD_SIZE - 1; _i -= 1) {
```

Comment:
I tested with using unchecked for both the loops and the tests pass, we can also do for the helper contract
do you see a risk doing it here ?